zerver/forms.py

```diff
@@ -80,13 +81,13 @@ def check_subdomain_available(subdomain: str, from_management_command: bool=Fals
         raise ValidationError(error_strings['extremal dash'])
     if not re.match('^[a-z0-9-]*$', subdomain):
         raise ValidationError(error_strings['bad character'])
-    if from_management_command:
+    if allow_reserved_subdomain:
         return
     if len(subdomain) < 3:
         raise ValidationError(error_strings['too short'])
     if is_reserved_subdomain(subdomain) or \
        Realm.objects.filter(string_id=subdomain).exists():
-        raise ValidationError(error_strings['unavailable'])
+        raise CommandError(error_strings['use flag'])
```

Comment:
This logic is wrong -- the `allow_reserved_subdomain` logic should only affect the  `is_reserved_subdomain` check (and maybe the length one) -- the exists check should not be possible to override.

Reply:
Ok I would change .